tensorflow_model_optimization/g3doc/guide/clustering/clustering_with_keras.ipynb

```diff
@@ -0,0 +1,1164 @@
+{
```

Comment:
<p>From the <a href="https://github.com/tensorflow/model-optimization/pull/257" target="_blank">consistent user experience RFC</a>, only the "cluster all layers" will be demonstrated in this end-to-end example. For motivation, see the RFC. The "cluster some layers" will instead be in the comprehensive guide. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/model-optimization/pull/330/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Reply:
<p>Agreed in principle. However, saw the 'benefits' in the RFC you mentioned (End-to-end example in Colab: demonstrate the benefits and end-to-end-path to) and had a second thought. </p><p>The benefit of using clustering on a model is: compressing the model while maintaining reasonable accuracy. The case to best demonstrate this in an end-to-end fashion should be selective clustering rather than the whole model clustering, which is more of a case for highest compression rate. </p><p>What do you think?</p>

Reply:
If you see that you will find better compression results on the current model for MNIST, showing selective clustering instead of clustering the whole model is reasonable to me. It'll be slightly more complicated, but I think there should be a way to keep it concise and simple. 

Otherwise, as a backup, people will eventually see in the Official Models samples that they need to do selective clustering in a lot of cases.

Reply:
<p>As discussed yesterday in the meeting, we will keep the full clustering case in the end-to-end example.</p>